verifiable-db/src/query/batching/row_chunk/mod.rs

```diff
@@ -0,0 +1,322 @@
+//! row chunks. A row chunk is a set of rows that have already been aggregated
+//! and whose rows are all proven to be consecutive. The first and last rows in
+//! the chunk are labelled as the `left_boundary_row` and the `right_boundary_row`,
+//! respectively, and are the rows employed to aggregate 2 different chunks.
+
+use mp2_common::{
+    serialization::circuit_data_serialization::SerializableRichField,
+    utils::{FromTargets, HashBuilder, SelectTarget, ToTargets},
+};
+use plonky2::{
+    hash::hash_types::{HashOutTarget, NUM_HASH_OUT_ELTS},
+    iop::target::{BoolTarget, Target},
+    plonk::circuit_builder::CircuitBuilder,
+};
+
+use crate::query::{
+    merkle_path::{MerklePathWithNeighborsTarget, NeighborInfoTarget},
+    universal_circuit::universal_query_gadget::UniversalQueryOutputWires,
+};
+
+/// This module contains gadgets to aggregate 2 different row chunks
+pub(crate) mod aggregate_chunks;
+/// This module contains gadgets to enforce whether 2 rows are consecutive
+pub(crate) mod consecutive_rows;
+
+/// Data structure containing the wires representing the data realted to the node of
+/// the row/index tree containing a row that is on the boundary of a row chunk.
+#[derive(Clone, Debug)]
+pub(crate) struct BoundaryRowNodeInfoTarget {
+    /// Hash of the node storing the row in the row/index tree
+    pub(crate) end_node_hash: HashOutTarget,
+    /// Data about the predecessor of end_node in the row/index tree
+    pub(crate) predecessor_info: NeighborInfoTarget,
+    /// Data about the predecessor of end_node in the row/index tree
+    pub(crate) successor_info: NeighborInfoTarget,
+}
+
+impl<'a, const MAX_DEPTH: usize> From<&'a MerklePathWithNeighborsTarget<MAX_DEPTH>>
+    for BoundaryRowNodeInfoTarget
+where
+    [(); MAX_DEPTH - 1]:,
+{
+    fn from(value: &'a MerklePathWithNeighborsTarget<MAX_DEPTH>) -> Self {
+        Self {
+            end_node_hash: value.end_node_hash,
+            predecessor_info: value.predecessor_info.clone(),
+            successor_info: value.successor_info.clone(),
+        }
+    }
+}
+
+impl SelectTarget for BoundaryRowNodeInfoTarget {
+    fn select<F: SerializableRichField<D>, const D: usize>(
+        b: &mut CircuitBuilder<F, D>,
+        cond: &BoolTarget,
+        first: &Self,
+        second: &Self,
+    ) -> Self {
+        Self {
+            end_node_hash: b.select_hash(*cond, &first.end_node_hash, &second.end_node_hash),
+            predecessor_info: NeighborInfoTarget::select(
+                b,
+                cond,
+                &first.predecessor_info,
+                &second.predecessor_info,
+            ),
+            successor_info: NeighborInfoTarget::select(
+                b,
+                cond,
+                &first.successor_info,
+                &second.successor_info,
+            ),
+        }
+    }
+}
+
+impl FromTargets for BoundaryRowNodeInfoTarget {
+    const NUM_TARGETS: usize = NUM_HASH_OUT_ELTS + 2 * NeighborInfoTarget::NUM_TARGETS;
```

Comment:
ok that's the reason why you want to export this as const - sounds good, forget about my previous comments on this then.